subsys/bluetooth/host/audio/ascs.c

```diff
@@ -674,22 +676,40 @@ static int ase_config(struct bt_ascs *ascs, struct bt_ascs_ase *ase,
 			}
 			stream = ase->ep.stream;
 		} else {
-			stream = bt_audio_stream_config(ascs->conn, &ase->ep,
-							cap, &ase->ep.codec);
-			if (stream == NULL) {
-				BT_ERR("Config failed");
+			stream = NULL;
+			if (server_cb != NULL && server_cb->config != NULL) {
+				err = server_cb->config(ascs->conn, &ase->ep,
+							cap, &ase->ep.codec,
+							&stream);
+			} else {
+				err = -EACCES;
```

Comment:
Wondering this is the best error code.
Alternatives: ENOENT?  ENODEV? Or maybe ENOSYS - function not implemented?

Reply:
I wondered that too. 

Could also be ECHILD, ESRCH or EOPNOTSUPP. 

I also considered the approach where we would silently just accept in case that no callback was provided (except for the one where we need a stream from the callback). 

What's your opinion?

Reply:
In media_proxy we silently ignore - if there is no callback registered, there will be no callback.
If it is important to have the callbacks in place, one might argue that it should not be possible to register empty callbacks, or run without registering callbacks, but we do not du that.
No strong opinion, really, except that I prefer e.g EOPNOTSUPP to ENOACCES

Reply:
The difference between those callbacks and the ones here, is that the ones in media_proxy is informative, and these are requesting data. 

An alternative that is used by gatt.c is having an __ASSERT for the callbacks. 

So we have
1) Silently accept (possibly with log)
2) Always reject  (possibly with log)
3) __ASSERT and require callbacks to be set when registering the callbacks (which is a separate function than actually registering the server).

Reply:
Updated with `EOPNOTSUPP` instead